src/OpenTelemetry.Api/Baggage/BaggageManagerBase.cs

```diff
@@ -0,0 +1,35 @@
+// <copyright file="BaggageManagerBase.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using OpenTelemetry.Api.Context.Propagation;
+
+namespace OpenTelemetry.Api.Baggage
+{
+    public abstract class BaggageManagerBase
+    {
+        public abstract string GetEntry(BaggageContext context, string key);
+
+        public abstract BaggageContext SetEntry(BaggageContext context, string key, string value);
+
+        public abstract BaggageContext RemoveEntry(BaggageContext context, string key);
+
+        public abstract BaggageContext ClearEntries(BaggageContext context);
+
+        public abstract HttpInjectorBase GetHttpInjector();
```

Comment:
Is there a requirement to propagate extractors/injectors through the manager class? Could it be done as a constructor argument to adapters instead?

How extractor/injections are going to be set up by users? On the tracer/meter factory?